backtester/eventhandlers/strategies/strategies_types.go

```diff
@@ -2,17 +2,17 @@ package strategies
 
 import (
 	"github.com/thrasher-corp/gocryptotrader/backtester/data"
-	"github.com/thrasher-corp/gocryptotrader/backtester/eventhandlers/portfolio"
 	"github.com/thrasher-corp/gocryptotrader/backtester/eventtypes/signal"
+	"github.com/thrasher-corp/gocryptotrader/backtester/funding"
 )
 
 // Handler defines all functions required to run strategies against data events
 type Handler interface {
 	Name() string
 	Description() string
-	OnSignal(data.Handler, portfolio.Handler) (signal.Event, error)
```

Comment:
How is the strategy supposed to know whether it's already in a trade if it doesn't have access to the portfolio?